test/Test/Data/String/CodePoints.purs

```diff
@@ -0,0 +1,185 @@
+module Test.Data.String.CodePoints (testStringCodePoints) where
+
+import Prelude (Unit, discard, negate, (==), ($), (&&), (<), (<$>))
+
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE, log)
+
+import Data.Maybe (Maybe(..), isNothing, maybe)
+import Data.String.CodePoints
+
+import Test.Assert (ASSERT, assert)
+
+str :: String
+str = "a\xDC00\xD800\xD800\x16805\x16A06\&z"
+
+testStringCodePoints :: forall eff. Eff (console :: CONSOLE, assert :: ASSERT | eff) Unit
+testStringCodePoints = do
+  log "codePointAt"
+  assert $ codePointAt (-1) str == Nothing
+  assert $ codePointAt 0 str == (codePointFromInt 0x61)
+  assert $ codePointAt 1 str == (codePointFromInt 0xDC00)
+  assert $ codePointAt 2 str == (codePointFromInt 0xD800)
+  assert $ codePointAt 3 str == (codePointFromInt 0xD800)
+  assert $ codePointAt 4 str == (codePointFromInt 0x16805)
+  assert $ codePointAt 5 str == (codePointFromInt 0x16A06)
+  assert $ codePointAt 6 str == (codePointFromInt 0x7A)
+  assert $ codePointAt 7 str == Nothing
+
+  log "count"
+  assert $ count (\_ -> true) "" == 0
+  assert $ count (\_ -> false) str == 0
+  assert $ count (\_ -> true) str == 7
+  assert $ count (\x -> codePointToInt x < 0xFFFF) str == 4
+  assert $ count (\x -> codePointToInt x < 0xDC00) str == 1
+
+  log "drop"
+  assert $ drop (-1) str == str
+  assert $ drop 0 str == str
+  assert $ drop 1 str == "\xDC00\xD800\xD800\x16805\x16A06\&z"
+  assert $ drop 2 str == "\xD800\xD800\x16805\x16A06\&z"
+  assert $ drop 3 str == "\xD800\x16805\x16A06\&z"
+  assert $ drop 4 str == "\x16805\x16A06\&z"
+  assert $ drop 5 str == "\x16A06\&z"
+  assert $ drop 6 str == "z"
+  assert $ drop 7 str == ""
+  assert $ drop 8 str == ""
+
+  log "dropWhile"
+  assert $ dropWhile (\c -> true) str == ""
```

Comment:
Possibly better to use `const true` or `\_ -> true` for when we get unused variable warnings?